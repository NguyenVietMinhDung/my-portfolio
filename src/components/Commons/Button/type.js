export type Props = {
  text: string,
  handleClickEvent: Function,
};
