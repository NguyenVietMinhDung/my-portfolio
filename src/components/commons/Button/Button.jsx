/* eslint-disable jsx-a11y/no-static-element-interactions */
// @flow
import React, { memo } from 'react';
import CSSModules from 'react-css-modules';
import styles from './Button.scss';
import type { Props } from './type';

const Button = (props: Props) => {
  const { text, handleClickEvent } = props;
  return (
    <div
      styleName="btn"
      onClick={handleClickEvent}
    >
      { text }
    </div>
  );
};

export default memo<Props>(CSSModules(Button, styles));
