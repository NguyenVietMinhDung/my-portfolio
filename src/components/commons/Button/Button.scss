@import '../../../styles/abstracts/variables';
@import '../../../styles/abstracts/mixins';

.btn {
	width: fit-content;
	padding: 0 40px;

	cursor: pointer;
	user-select: none;
	transition: all 0.3s ease-out 0s;

	border: 1px solid $violet-color;
	border-radius: 50px;
	background-color: $violet-color;
	box-shadow: 0 5px 8px 0 $violet-color;

	line-height: 48px;

	&:hover {
		color: $violet-color;
		background-color: $white-color;
	}

	@include font(inherit, uppercase, $white-color, 15px, 500);
}
