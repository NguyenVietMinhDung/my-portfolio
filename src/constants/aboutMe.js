export const TITLE = 'About Me';
export const GREETING = 'Hi There! I\'m Dung Nguyen';
export const SUMMARY = 'I am an ambitious Web\n'
  + 'Application Developer who\n'
  + 'always tries to improve himself\n'
  + 'everyday, a quick learner who\n'
  + 'can absorb new ideas and can\n'
  + 'communicate clearly and\n'
  + 'effectively. Looking for an\n'
  + 'ambitious company which will\n'
  + 'challenge my developer and\n'
  + 'allows me to continue\n'
  + 'to develop my knowledge and\n'
  + 'potential.';
export const basicInfo = [
  {
    iconClassName: 'lni-calendar',
    name: 'Date of birth',
    value: '16 January 1995',
  },
  {
    iconClassName: 'lni-envelope',
    name: 'Email',
    value: 'ngvietminhdung@gmail.com',
  },
  {
    iconClassName: 'lni-phone-handset',
    name: 'Phone',
    value: '(076) 401-2912',
  },
  {
    iconClassName: 'lni-map-marker',
    name: 'Location',
    value: 'Tan Binh District, HCM City',
  },
];
export const skills = [
  {
    itemId: 'skill-1',
    name: 'JavaScript, HTML, CSS',
    value: 80,
  },
  {
    itemId: 'skill-2',
    name: 'ReactJS',
    value: 60,
  },
  {
    itemId: 'skill-3',
    name: 'SCSS',
    value: 60,
  },
  {
    itemId: 'skill-5',
    name: 'Webpack',
    value: 50,
  },
  {
    itemId: 'skill-6',
    name: 'Node.js',
    value: 60,
  },
  {
    itemId: 'skill-7',
    name: 'Git',
    value: 60,
  },
];
