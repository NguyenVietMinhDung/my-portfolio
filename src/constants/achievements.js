export const TITLE = 'Achievements';
export const timeline = {
  iconClassName: 'lni-certificate',
  items: [
    {
      key: 1,
      organization: 'Amazon Web Services (AWS)',
      startDate: 'August 2020',
      endDate: 'August 2023',
      description: 'AWS Certified Cloud Practitioner',
    },
  ],
};
