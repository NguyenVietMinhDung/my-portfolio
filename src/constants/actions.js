export const ACTIVATE_NAVIGATION_ITEM = 'ACTIVATE_NAVIGATION_ITEM';
export const OPEN_NAVIGATION_MENU = 'OPEN_NAVIGATION_MENU';
