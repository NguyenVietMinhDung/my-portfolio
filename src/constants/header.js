export const NAME = 'Dung Nguyen';
export const DESCRIPTION = 'A Freelance UI Designer & Web Developer';
export const BUTTON_TEXT = 'DOWNLOAD MY CV';
export avatarUrl from '../assets/images/avatar.png';
