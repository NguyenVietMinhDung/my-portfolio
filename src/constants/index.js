import * as actions from './actions';
import * as app from './app';
import * as navbar from './navbar';
import * as header from './header';
import * as footer from './footer';
import * as aboutMe from './aboutMe';
import * as experiences from './experiences';
import * as educations from './educations';
import * as projects from './projects';
import * as achievements from './achievements';

export default {
  actions,
  app,
  navbar,
  header,
  footer,
  aboutMe,
  experiences,
  educations,
  projects,
  achievements,
};
