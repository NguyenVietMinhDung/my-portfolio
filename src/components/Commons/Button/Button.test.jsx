import React from 'react';
import { shallow } from 'enzyme';
import { BUTTON_TEXT } from '../../../constants/header';
import Button from './Button';

const defaultProps = {
  text: BUTTON_TEXT,
  handleClickEvent: jest.fn(),
};

describe('<Button />', () => {
  test('renders correctly', () => {
    const component = shallow(<Button {...defaultProps} />);
    expect(component).toMatchSnapshot();
  });
});
