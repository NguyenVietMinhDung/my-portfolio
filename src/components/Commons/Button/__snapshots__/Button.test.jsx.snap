// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders correctly 1`] = `
<div
  className="btn"
  onClick={[MockFunction]}
>
  DOWNLOAD MY CV
</div>
`;
